utils/zerovec/design_doc.md

```diff
@@ -33,6 +33,8 @@ It is worth noting that [`rkyv`](https://docs.rs/rkyv) satisfies many of these r
 3. Limited support for data overrides (mixing owned runtime data with borrowed static data)
 4. Little-endian and big-endian require different data files, meaning that the flag to toggle between them also needs to percolate through the data system
 
+Relative to `rkyv`, the primary limitation of `zerovec` is its inability to produce data aligned to anything other than 1 byte; i.e., there is no way for it to represent something like `&[u32]`. However, we have [benchmarked](https://github.com/unicode-org/icu4x/pull/1391) that for single-element read operations, the performance impact of an unaligned read is minimal. See [this issue](https://github.com/unicode-org/icu4x/issues/1426#issuecomment-1045043829) for a discussion of options in situations where alignment has a measurable performance benefit (perhaps vectorized operations like matrix multiplication).
```

Comment:
Praise: Adding this comment is a good idea.

Nit/follow-up: we've benchmarked x86-64. How much work would it be to run the same benchmarks on an arm machine? Arm used to be much stricter about alignment, although I think it's gotten better. It would be nice to double-check, though.

Reply:
I spent about half an hour last month trying to get Criterion to cross-compile to Android to run on my phone but wasn't successful. I don't have access to another ARM machine. It should be quite easy for someone who has access though to run the benchmark.

Reply:
ARM discussion:

https://github.com/unicode-org/icu4x/pull/1391#issuecomment-1250167435